### PR TITLE
Move from CMake to Meson build

### PR DESCRIPTION
**Pull request by abinayaddhandapani**
Move compilation from CMake to meson build.
format the code using clang-format
Remove unused header files included in compilation

### CODE REVIEW

File: src/Config.cpp [hunk 1]
```diff
@@ -16,63 +16,28 @@ uint16_t Configuration::PcieAerPollingPeriod;
...
-    {"P1_DIMM_I", "null"},
-    {"P1_DIMM_I1", "null"},
-    {"P1_DIMM_J", "null"},
-    {"P1_DIMM_J1", "null"},
-    {"P1_DIMM_K", "null"},
-    {"P1_DIMM_K1", "null"},
-    {"P1_DIMM_L", "null"},
-    {"P1_DIMM_L1", "null"}
-};
+    {{"P1_DIMM_A", "null"},  {"P1_DIMM_A1", "null"}, {"P1_DIMM_B", "null"},
```
Comment (supven01) [inline src/Config.cpp:33]: Why the wrap around?

File: src/Config.cpp [hunk 2]
```diff
@@ -16,63 +16,28 @@ uint16_t Configuration::PcieAerPollingPeriod;
...
+     {"P1_DIMM_B1", "null"}, {"P1_DIMM_C", "null"},  {"P1_DIMM_C1", "null"},
```
Comment (supven01) [inline src/Config.cpp:34]: Why the wrap around?

**Review by supven01 — CHANGES_REQUESTED** [review]
Add a verbose commit message to indicate what all changes were done. I am pretty sure its more than just the move from CMake to Meson build

File: src/Config.cpp (hunk 1 above)
Comment (abinayaddhandapani) [reply src/Config.cpp:33]: The clang formatter has formatted the P0_DimmLabels

File: src/Config.cpp (hunk 2 above)
Comment (abinayaddhandapani) [reply src/Config.cpp:34]: The clang formatter has formatted the P1_DimmLabels

File: src/Config.cpp (hunk 1 above)
Comment (supven01) [reply src/Config.cpp:33]: ok.

**Pull request merged**